app_feup/lib/controller/load_static/terms_and_conditions.dart

```diff
@@ -1,14 +1,29 @@
 import 'dart:convert';
 
+import 'package:connectivity/connectivity.dart';
 import 'package:crypto/crypto.dart';
+import 'package:flutter/cupertino.dart';
 import 'package:flutter/services.dart' show rootBundle;
+import 'package:http/http.dart' as http;
 import 'package:uni/controller/local_storage/app_shared_preferences.dart';
 
 Future<String> readTermsAndConditions() async {
   try {
+    if (await (Connectivity().checkConnectivity()) != ConnectionState.none) {
```

Comment:
At least in the emulator, checkConnectivity() is always returning different from ConnectionState.none, even if both the Wi-Fi and mobile data are turned off, so it's not falling back to locally stored T&C.

Reply:
This seems to be a bit harder to test in the emulator but a means to do so is to activate airplane mode!